### DISCUSSION TITLE
Alpha and Default AOV outputs

### DISCUSSION THREAD

**Discussion by Kolupsy** [bnpr/Malt#180] [Feedback & Feature Requests]
When using the CustomA-D outputs on malt there are two 'issues' I encountered.
1. The default value for all the Custom outputs is (0,0,0,0) but only if enabled. I think it should also be (0,0,0,0) when not enabled/overridden. The reasoning behind it being that I would define the Alpha channels as sort of the Mask for the custom output, so when the Custom outputs are not being used, they should be excluded from the mask and not use (1,1,1,1) as a value.

![image](https://user-images.githubusercontent.com/88381426/143585582-662942a5-1ba8-4a8b-adec-42f164d2e23e.png)

2. If an object has a material with Transparency enabled and PO.color has 0 alpha, any custom outputs wont work, so I cannot mask objects from the main color pass while still keeping the custom outputs enabled.

**Comment by pragma37** [comment]
1. Yes, this is kind of weird. I have to look into it but I don't consider it a high priority.

2. That's by design. In OpenGL (or anywhere else, AFAIK) you can't discard just one of the Shader Outputs.
I'm currently working on adding Scene Passes to Render Layer nodes:
https://github.com/bnpr/Malt/discussions/148#discussioncomment-1478231
That should let you achieve what you want.
